src/transformers/models/rag/retrieval_rag.py

```diff
@@ -372,8 +365,7 @@ class RagRetriever:
 
     def __init__(self, config, question_encoder_tokenizer, generator_tokenizer, index=None, init_retrieval=True):
         self._init_retrieval = init_retrieval
-        requires_datasets(self)
-        requires_faiss(self)
+        requires_backends(self, ["datasets", "faiss"])
```

Comment:
This kind of refactor is nice and will give one clear error message containing all required dependencies.

Reply:
This is very nice indeed